hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java

```diff
@@ -218,12 +218,17 @@ public static OzoneQuota parseNameSpaceQuota(String quotaInNamespace) {
       throw new IllegalArgumentException(
           "Quota string cannot be null or empty.");
     }
-    long nameSpaceQuota = Long.parseLong(quotaInNamespace);
-    if (nameSpaceQuota <= 0) {
-      throw new IllegalArgumentException(
-          "Invalid values for namespace quota: " + nameSpaceQuota);
+    try {
+      long nameSpaceQuota = Long.parseLong(quotaInNamespace);
+      if (nameSpaceQuota <= 0) {
+        throw new IllegalArgumentException(
+            "Invalid value for namespace quota: " + nameSpaceQuota);
+      }
+      return new OzoneQuota(nameSpaceQuota, new RawQuotaInBytes(Units.B, -1));
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException(quotaInNamespace + " is invalid. " +
+          "The quota value should be a positive integer.");
```

Comment:
I think this should be "positive long" instead of "positive integer".

Reply:
Changed to "positive long"

Reply:
I don't know. What is a "long" to someone who is not a programmer? Most people know what an "integer" is, but a long is a programmer term and not well understood, hence my suggestion to change it to integer for the purposes of an error message. "Number" would have been better, but that does not run out decimal points etc.

Reply:
I think we should change in other places also like one in PR description to make it consistent.

`bash-4.2$ ozone sh volume create vol1 --namespace-quota test
Invalid values for quota, to ensure that the Quota format is legal(supported values are positive long values).
`

Reply:
Maybe we can term it to integer, specifying the limits in brackets just as space-quota error [message](https://github.com/apache/ozone/blob/43cdb74fc455430d5f3fd72026b617cbb1a2bd47/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java#L197).

Reply:
Integer with limits makes sense. 

If this same problem exists in other commands, it would be good to clean it all up in one go here if we can, and make it all consistent?

Reply:
I think we can update `namespace-quota` and `space-quota` which is handled in this PR. Also please update in PR description with the new error message.

Reply:
Suggested error message :
-  space-quota 
`<quota-value> is invalid. The quota value should be a positive integer(Long.MAX_VALUE) with byte numeration(B, KB, MB, GB and TB)`
- namespace-quota
`<quota-value> is invalid. The quota value should be a positive integer(Long.MAX_VALUE)`

We can make this as the error messages for respective quotas.

Reply:
Maybe:

```
<quota-value> is invalid. The quota value should be a positive integer between 1 and Long.MAX_VALUE
```
Is slightly better.

Reply:
Implemented the latest discussed message and related tests. Please review